include/alpaka/core/OmpSchedule.hpp

```diff
@@ -0,0 +1,94 @@
+/* Copyright 2020 Sergei Bastrakov
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/Common.hpp>
+
+#ifdef _OPENMP
+#    include <omp.h>
+#endif
+
+#include <cstdint>
+
+
+namespace alpaka
+{
+    namespace omp
+    {
+        //#############################################################################
+        //! Representation of OpenMP schedule information: kind and chunk size.
+        //!
+        //! This class can be used regardless of whether OpenMP is enabled
+        struct Schedule
+        {
+            //#############################################################################
+            //! Mirror of the OpenMP enum omp_sched_t with the corresponding integer values
+            enum Kind
+            {
+                Static = 1,
+                Dynamic = 2,
+                Guided = 3,
+                Auto = 4
+            };
+
+            //! Integer representation of schedule kind.
+            //!
+            //! Not stored as Kind to reduce casts when converting to and from omp_sched_t
+            uint32_t kind;
+
+            //! Chink size.
+            //!
+            //! Same as in OpenMP, value 0 corresponds to default chunk size
+            int chunkSize;
+
+            constexpr Schedule(Kind kind = Guided, int chunkSize = 0)
+                : kind(static_cast<uint32_t>(kind))
```

Comment:
There are few errors about name shadowing errors: https://github.com/alpaka-group/alpaka/pull/1223/checks?check_run_id=1464801629#step:5:4290 with `kind` and `chunkSize`.
In PIConGU we use `m_varname` in such case for class member and the human readable name `varname` for method parameter to expose nice names in the userinterface.

Reply:
I don't think it actually shadows anything with empty constructor body, but will fix. Here addintg `m_` is not very nice, as this is POD, and so public members are part of interface.